llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -54,9 +76,16 @@ TGLexer::TGLexer(SourceMgr &SM, ArrayRef<std::string> Macros) : SrcMgr(SM) {
   PrepIncludeStack.push_back(
       std::make_unique<std::vector<PreprocessorControlDesc>>());
 
-  // Put all macros defined in the command line into the DefinedMacros set.
-  for (const std::string &MacroName : Macros)
+  // Add all macros defined on the command line to the DefinedMacros set.
+  // Check invalid macro names and print fatal error if we find one.
+  for (StringRef MacroName : Macros) {
+    const char *End = lexMacroName(MacroName);
+    if (End != MacroName.end())
+      PrintFatalError("Invalid macro name `" + MacroName +
+                      "` specified on command line.");
```

Comment:
'.' is not canonical in error messages

Reply:
Also, the message should start with lowercase letter.

Reply:
All PrintError() instances in this file have their error message starting with upper case.

Reply:
I've removed the .

Reply:
Ah, ok then.